### PR TITLE
Mic 5211/cron job

### PR DESCRIPTION
**Pull request by albrja**
### Slack test channel if not main/cron job
- *Category*: 
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5211

### Changes and notes
-add vivarium cron job Jenkins pipeline

### CODE REVIEW

**Comment by albrja** [comment]
This will slack the test channel if the branch is not main - meaning the cron jobs will be the only ones to slack the CI channel

**Review by hussain-jafari — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -204,10 +204,22 @@ pipeline {
...
+                            tokenCredentialId: "slack"
+                } else {
+                  echo "This is branch ${GIT_BRANCH}. Sending a failure message to Slack."
+                  // TODO: DM the developer instead of the slack channel
+                  slackSend channel: "#simsci-ci-status-test",
+                            message: ":x: JOB FAILURE: $JOB_NAME - $BUILD_ID\n\n${BUILD_URL}console",
+                            teamDomain: "ihme",
+                            tokenCredentialId: "slack"
+                }
+              }
```
Comment (rmudambi) [inline Jenkinsfile:222]: Aren't these blocks identical other than the name of the channel? Can't you do
```groovy
if (${GIT_BRANCH} == "main") {
    channelName = "#${params.SLACK_TO}"
} else {
    channelName = "#simsci-ci-status-test"
}
echo ...
slackSend channel: channelName,
    ...
```

Comment (albrja) [reply Jenkinsfile:220]: Made this change.

File: Makefile [hunk 2]
```diff
@@ -56,6 +56,11 @@ build-env: # Make a new conda environment
 install: # Install setuptools, install this package in editable mode
 	pip install --upgrade pip setuptools
 	pip install -e .[DEV]
+	@echo "Checking if the Layered Config Tree repository has the branch $(GIT_BRANCH)..."
+	@$(eval BRANCH_EXISTS=$(shell curl -s https://api.github.com/repos/ihmeuw/layered_config_tree/branches | grep -q '"name": "$(GIT_BRANCH)"' && echo "yes" || echo "no"))
+	@if [ "$(BRANCH_EXISTS)" = "yes" ]; then \
```
Comment (rmudambi) [inline Makefile:61]: Is it possible to use the bash script I created earlier this week to check for and install branches?

Comment (albrja) [reply Makefile:61]: I was just thinking about this. I can play around with it and see

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -45,7 +45,7 @@ jobs:
           cat ../vivarium_build_utils/install_dependency_branch.sh
           echo ""
           echo "----------------------------------------"
-          sh ../vivarium_build_utils/install_dependency_branch.sh layered_config_tree ${branch_name}
+          sh ../vivarium_build_utils/install_dependency_branch.sh layered_config_tree ${branch_name} jenkins
```
Comment (rmudambi) [inline .github/workflows/build.yml:48]:
```suggestion
          sh ../vivarium_build_utils/install_dependency_branch.sh layered_config_tree ${branch_name} github
```

File: Makefile [hunk 4]
```diff
@@ -56,6 +57,20 @@ build-env: # Make a new conda environment
...
+		git clone https://github.com/ihmeuw/vivarium_build_utils.git; \
+	else \
+		echo "vivarium_build_utils already exists. Skipping clone."; \
+	fi
+	@echo "Contents of install_dependency_branch.sh"
+	@echo "----------------------------------------"
+	@cat vivarium_build_utils/install_dependency_branch.sh
+	@echo ""
+	@echo "----------------------------------------"
+	@sh vivarium_build_utils/install_dependency_branch.sh layered_config_tree ${GIT_BRANCH}
```
Comment (rmudambi) [inline Makefile:73]:
```suggestion
	@sh vivarium_build_utils/install_dependency_branch.sh layered_config_tree ${GIT_BRANCH} jenkins
```

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (albrja) [reply .github/workflows/build.yml]: Sorry I had made these but hadn't pushed them.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**